src/components/AddUsersModal/index.jsx

```diff
@@ -0,0 +1,328 @@
+import { Button, Icon, Modal } from '@edx/paragon';
+
+import H3 from '../H3';
+import TextAreaAutoSize from '../TextAreaAutoSize';
+import StatusAlert from '../StatusAlert';
+import FileInput from '../FileInput';
+import { validateEmailTemplateFields, validateEmailAddresses } from '../../utils';
+
+class AddUsersModal extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.errorMessageRef = React.createRef();
+    this.modalRef = React.createRef();
+
+    this.state = {
+      fields: {
+        'email-addresses': null,
+      },
+    };
+
+    this.validateFormData = this.validateFormData.bind(this);
+    this.handleModalSubmit = this.handleModalSubmit.bind(this);
+    this.handleFieldOnChange = this.handleFieldOnChange.bind(this);
+  }
+
+  componentDidMount() {
+    const {
+      enterpriseId,
+      fetchSubscriptionDetails,
+    } = this.props;
+
+    if (enterpriseId) {
+      fetchSubscriptionDetails(enterpriseId);
+    }
+
+    const { current: { firstFocusableElement } } = this.modalRef;
+
+    if (firstFocusableElement) {
+      firstFocusableElement.focus();
+    }
+  }
+
+  componentDidUpdate(prevProps) {
+    const {
+      submitFailed,
+      submitSucceeded,
+      onClose,
+      error,
+    } = this.props;
+
+    const errorMessageRef = this.errorMessageRef && this.errorMessageRef.current;
+
+    if (submitSucceeded && submitSucceeded !== prevProps.submitSucceeded) {
+      onClose();
+    }
+
+    if (submitFailed && error !== prevProps.error && errorMessageRef) {
+      // When there is an new error, focus on the error message status alert
+      errorMessageRef.focus();
+    }
+  }
+
+  componentWillUnmount() {
+    // Clear the subscription data
+    this.props.clearSubscriptionDetails();
+  }
+
+  validateFormData(formData) {
+    const userEmailsKey = 'email-addresses';
+    const emailsCSVKey = 'csv-email-addresses';
+    const emailTemplateKey = 'email-template-body';
+
+    /* eslint-disable no-underscore-dangle */
+    const errors = validateEmailTemplateFields(formData);
+
+    if (!formData[emailTemplateKey]) {
+      const message = 'An email template is required.';
+      errors[emailTemplateKey] = message;
+      errors._error.push(message);
+    }
+
+    if (!formData[userEmailsKey] && !formData[emailsCSVKey]) {
+      const message = 'Either user emails or emails csv must be provided.';
+      errors[userEmailsKey] = message;
+      errors[emailsCSVKey] = message;
+      errors._error.push(message);
+    }
+
+    const textAreaEmails = formData[userEmailsKey] && formData[userEmailsKey].split(/\r\n|\n/);
+    const csvEmails = formData[emailsCSVKey];
+
+    let {
+      invalidEmailIndices,
+    } = textAreaEmails ? validateEmailAddresses(textAreaEmails) : validateEmailAddresses(csvEmails);
+
+    // 1 is added to every index to fix off-by-one error in messages shown to the user.
+    invalidEmailIndices = invalidEmailIndices.map(i => i + 1);
```

Comment:
can we abstract this inside `validateEmailAddresses` otherwise we need to do this every time we use `validateEmailAddresses`